paperqa/llms.py

```diff
@@ -149,6 +152,40 @@ async def embed_documents(self, client, texts):
         return np.concatenate(all_embeds, axis=1)
 
 
+class VoyageAIEmbeddingModel(EmbeddingModel):
+    """A wrapper around Voyage AI's client lib."""
+
+    name: str = Field(default="voyage-large-2")
+    embedding_type: str = Field(default="document")
+    batch_size: int = 10
+
+    def set_mode(self, mode: str):
+        if mode == "query":
+            self.embedding_type = "query"
+        elif mode == "document":
+            self.embedding_type = "document"
+        else:
+            raise NotImplementedError(
+                f"Mode {mode} is not implemented for VoyageAI embeddings."
+            )
```

Comment:
Getting a code smell for an `Enum` with a method

Reply:
Thought that too, but this will be different per embedding model, Voyage uses 'query'/'document', but not some of the other embeddings on the MTEB. I guess we could force a mapping to these two values.

Reply:
I'll swap this out with an enum actually -- just scope it for this single model.